media/CircleEditor/view-sidebar.js

```diff
@@ -414,13 +369,104 @@ sidebar.EditAttributesView = class {
     }
 };
 
-/**
- * TODO Implement sidebar.EditInputsView Class
- */
+sidebar.EditInputsView = class {
+    constructor(host, inputs, isCustom, node) {
+        this._host = host;
+        this._elements = [];
+        this._inputs = [];
+        this._index = 0;
+        this._isCustom = isCustom;
+        this._node = node;
 
-/**
- * TODO Implement sidebar.EditOutputsView Class
- */
+        this._addHeader('Inputs');
+        for (const input of inputs) {
+            this._addInput(input.name, input);
+            this._index++;
+        }
+    }
+
+    _addHeader(title) {
+        const headerElement = this._host.document.createElement('div');
+        headerElement.className = 'sidebar-view-header';
+        headerElement.innerText = title;
+        this._elements.push(headerElement);
+    }
+
+    _addInput(name, input) {
+        if (input.arguments.length > 0) {
+            const inputAttributes = {
+                title: 'input',
+                index: this._index,
```

Comment:
Adding and reading `_index` value are done at difference functions.
This is not good for maintaining because we can miss the overall logic easily.
So following would be better for redability.

```suggestion
                index: this._inputs.length,
```

`sidebar.EditOutputsView` is same :)